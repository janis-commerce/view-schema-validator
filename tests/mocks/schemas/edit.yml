service: sac
name: claim-motive-edit
root: Edit
source:
  service: sac
  namespace: claim-motive
  method: get
  resolve: false
header:
  title:
    beforeId:
      - name: test
        component: BadgeLetter
    afterId:
      - name: test
        component: StatusChip
        mapper: translate
        componentAttributes:
          useTheme: true
      - name: test2
        component: StatusChip
        componentAttributes:
          useTheme: themeOne
themes:
  themeOne:
    new: black
    closed: green
    _default: grey
  themeTwo:
    new: grey
    closed: fizzgreen
sections:
- name: mainFormSection
  rootComponent: MainForm
  topComponents:
    - component: TestComponent
    - component: TestComponentLeft
      position: left
    - component: TestComponentRight
      position: right
    - component: ActionButtons
      position: right
      actions:
        - name: new
          icon: star_light
          color: fizzGreen
          type: link
          options:
            path: /service/namespace/new

        - name: testAction
          icon: star_light
          color: fizzGreen
          type: endpoint
          options:
            endpoint:
              service: sac
              namespace: claim
              method: get
              resolve: false
            endpointParameters:
              id: id

        - name: testAction2
          icon: star_light
          color: fizzGreen
          type: endpoint
          options:
            endpoint:
              service: sac
              namespace: claim
              method: get
              resolve: false
            endpointParameters:
              - name: status
                target: path
                value:
                  dynamic: id
              - name: status
                target: query
                value:
                  static: 1

  fieldsGroup:
  - name: detail
    position: left
    icon: catalogue
    collapsible: true
    fields:
    - name: name
      component: Select
      componentAttributes:
        translateLabels: true
        labelFieldName: motiveName
        labelPrefix: common.status.
        options:
          scope: local
          valuesMapper:
            label: name
            value: id
          values:
            - label: active
              value: 1
            - label: inactive
              value: 0
      validations:
        - - name: required
        - - name: maxLength
            options:
              length: 50
      conditions:
        enableWhen:
          - - name: isEmpty
              field: name
            - name: isOneOf
              field: someField
              referenceValue:
                - test1
                - test2
        showWhen:
          - - name: isNotEmpty
              field:
                - test
                - name
            - name: isNotEqualTo
              field: name
              referenceValueType: static
              referenceValue: null
    - name: description
      component: Textarea
      validations:
        - - name: maxLength
            options:
              length: 50
      conditions:
        showWhen:
          - - name: isEqualTo
              field: user1
              referenceValueType: dynamic
              referenceValue: name
            - name: isNotOneOf
              field: someField
              referenceValue:
                - test1
                - test2
          - - name: isNotEqualTo
              field:
                - test
                - name
              referenceValue: true
    - name: appliesToLogistics
      component: Switch
      validations:
        - - name: required
    - name: missingStockDefault
      component: Switch
      validations:
        - - name: required
    - name: test
      component: Chip
    - name: user1
      component: UserChip
    - name: user2
      component: UserChip
      componentAttributes:
        userDataSource:
          email: email
          firstname: firstname
          lastname: lastname
          image: image
    - name: color
      component: ColorPicker

    - name: exampleCode
      component: Code

    - name: exampleCodeTwo
      component: Code
      componentAttributes:
        language: json
        canEdit: true

    - name: exampleMap
      component: Map
      componentAttributes:
        fieldsMapping:
          latitude: latitude
          longitude: longitude

    - name: exampleMapTwo
      component: Map
      componentAttributes:
        showSearchBar: true
        canAddMarkers: true
        maxMarkersQuantity: 5
        fieldsMapping:
          latitude: latitude
          longitude: longitude

    - name: exampleMapThree
      component: Map

    - name: asyncData
      component: AsyncWrapper
      componentAttributes:
        staticFilters:
        - name: status
          target: path
          value:
            dynamic: id
        - name: status
          target: query
          value:
            static: 1
        source:
          service: id
          namespace: user
          method: get
          resolve: false
        dataMapping:
          firstname: userFirstname
        field:
          name: userFirstname
          component: Text

    - name: linkTest
      component: Link
      componentAttributes:
        target: _blank
        translateLabels: true
        label: test.test.test
        labelField: asdasd

    - name: fieldTriggers
      component: Input
      triggers:
        - endpoint:
            service: sac
            namespace: claim
            method: list
            resolve: false
          endpointParameters:
            - name: status
              target: path
              value:
                dynamic: id
            - name: status
              target: query
              value:
                static: 1
          dataMapping:
            name: firstname
          triggerOnLoad: true
        - endpoint:
            service: sac
            namespace: claim
            method: list
            resolve: false
          dataMapping:
            name: firstname

    - name: fieldsArray
      component: FieldsArray
      componentAttributes:
        canChangeElements: true
        minElements: 1
        maxElements: 3
        fields:
          - name: test
            component: Text

    - name: fieldsArrayTwo
      component: FieldsArray
      componentAttributes:
        canChangeElements: true
        uniqueField: true
        minElements: 1
        fields:
          - name: test
            component: Text

    - name: template
      component: HTML
      componentAttributes:
        sourceField: template

    - name: emptyTemplate
      component: HTML

  - name: others
    fields:
      - name: status
        component: Select
        validations:
          - - name: required
        componentAttributes:
          translateLabels: true
          canClear: true
          options:
            scope: local
            values:
            - label: common.status.active
              value: 1
            - label: common.status.inactive
              value: 0

      - name: selectRemote
        component: Select
        validations:
          - - name: required
        componentAttributes:
          translateLabels: true
          options:
            scope: remote
            endpoint:
              service: sac
              namespace: claim
              method: list
              resolve: false
            initialValuesEndpoint:
              service: sac
              namespace: claim
              method: list
              resolve: false
            initialValuesFilterName: id
            initialValuesPathParam: name
            endpointParameters:
              - name: status
                target: path
                value:
                  dynamic: id
              - name: status
                target: query
                value:
                  static: 1
            valuesMapper:
              label: name
              value: id

      - name: selectRemoteTwo
        component: Select
        componentAttributes:
          translateLabels: true
          options:
            scope: remote
            endpoint:
              service: sac
              namespace: claim
              method: list
              resolve: false
            initialValuesEndpoint:
              service: sac
              namespace: claim
              method: list
              resolve: false
            initialValuesFilterName: id
            initialValuesPathParam: name
            valuesMapper:
              label:
                template: '{0} {1} - ({2})'
                fields:
                 - firstname
                 - lastname
                 - email
              value: id

      - name: dateTime
        component: DateTimePicker
        componentAttributes:
          selectDate: false
          selectTime: true
          selectRange: true
      - name: checklist
        component: CheckList
        componentAttributes:
          optionsSource:
            service: sac
            namespace: claim-motive
            method: list
            resolve: false
          sectionField: claimMotiveName
          groupField: statusName
          labelField: name
          valueField: id
          translateSectionLabel: true
          translateGroupLabel: false
          translateCheckboxLabel: false
      - name: multiInputExample
        component: MultiInput
        componentAttributes:
          labelsPrefix: common.test.
          labelPrefix: common.test.
          translateLabels: true
          requiredFields:
            - test
            - test2

- name: semaphoreBrowse
  rootComponent: BrowseSection
  topComponents:
    - component: TestComponent
      attributes:
        name: test
        sarasa: test23
    - component: TestComponent2
      attributes:
        name: test
        sarasa: test23
  source:
    service: sac
    namespace: claim-semaphore
    method: browse
    resolve: false
  staticFilters:
    - name: status
      target: path
      value:
        dynamic: statusId
    - name: status
      target: query
      value:
        static: 1
    - name: status
      target: query
      value:
        static:
          id: "string"
  fields:
    - name: id
      component: BoldText
      attributes:
        sortable: true
        isDefaultSort: true
      mapper: addHashtag
    - name: color
      component: Text
    - name: user
      component: UserChip
    - name: name
      component: MediumText
    - name: rangeLow
      component: Chip
      componentAttributes:
        icon: icon_test
        iconColor: red
        borderColor: red
    - name: rangeHigh
      component: Chip

    - name: colorTestOne
      component: Color

    - name: colorTestTwo
      component: Color
      componentAttributes:
        showCode: true
    - name: sacClaimChange
      component: SacClaimChange

    - name: alerts
      component: Chip
      attributes:
        sortable: true
      mapper: booleanToWord
      filter:
        component: Select
        componentAttributes:
          translateLabels: true
          options:
            - label: common.boolean.yes
              value: 1
            - label: common.boolean.no
              value: 0
    - name: status
      component: StatusChip
      attributes:
        isStatus: true
        sortable: true
      componentAttributes:
        colorSource: statusColor
      mapper: translate
      filter:
        component: Select
        componentAttributes:
          translateLabels: true
          options:
            - label: common.status.active
              value: 1
            - label: common.status.inactive
              value: 0
- name: logsBrowseSection
  rootComponent: LogsBrowseSection
- name: filesSection
  rootComponent: FilesSection
  fileUploadEndpoint:
    service: sac
    namespace: claim
    method: upload
    resolve: false
  fileRelationEndpoint:
    service: sac
    namespace: claim
    method: upload
    resolve: false
  fileListEndpoint:
    service: sac
    namespace: claim
    method: upload
    resolve: false
  fileDeleteEndpoint:
    service: sac
    namespace: claim
    method: upload
    resolve: false
  fileGetEndpoint:
    service: sac
    namespace: claim
    method: upload
    resolve: false
  filesTypes:
    - image/png
- name: orderItemsSection
  rootComponent: OrderItemsSection
  sourceField: items
  source:
    service: oms
    namespace: order
    method: get
    resolve: false
  staticFilters:
    - name: status
      target: path
      value:
        dynamic: statusId
    - name: status
      target: query
      value:
        static: 1
    - name: status
      target: query
      value:
        static:
          id: "string"

- name: anotherOrderItemsSection
  rootComponent: OrderItemsSection

- name: apiKeysSection
  rootComponent: ApiKeysSection
  parentIdField: user
  source:
    service: id
    namespace: user
    method: list-api-keys
    resolve: false
  createEndpoint:
    service: id
    namespace: user
    method: create-api-key
    resolve: false
  deleteEndpoint:
    service: id
    namespace: user
    method: delete-api-key
    resolve: false
- name: testFormSection
  rootComponent: FormSection
  source:
    service: sac
    namespace: claim-motive
    method: get
    resolve: false
  target:
    service: sac
    namespace: claim-motive
    method: save
    resolve: false
  fieldsGroup:
  - name: detail
    position: left
    icon: catalogue
    collapsible: true
    fields:
    - name: name
      component: Select
      componentAttributes:
        translateLabels: true
        labelFieldName: motiveName
        options:
          scope: local
          valuesMapper:
            label: name
            value: id
          values:
            - label: common.status.active
              value: 1
            - label: common.status.inactive
              value: 0
      validations:
        - - name: required
        - - name: maxLength
            options:
              length: 50