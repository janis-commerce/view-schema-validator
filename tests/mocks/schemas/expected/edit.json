{
    "service": "sac",
    "name": "claim-motive-edit",
    "root": "Edit",
    "source": {
        "service": "sac",
        "namespace": "claim-motive",
        "method": "get",
        "resolve": false
    },
    "target": {
        "service": "sac",
        "namespace": "claim-motive",
        "method": "save",
        "resolve": false
    },
    "sections": [
        {
            "name": "mainFormSection",
            "rootComponent": "MainForm",
            "actions": [
                {
                    "name": "new",
                    "icon": "star_light",
                    "color": "fizzGreen",
                    "type": "link",
                    "options": {
                        "path": "/service/namespace/new"
                    }
                }
            ],
            "topComponents": [
                {
                    "component": "TestComponent"
                }
            ],
            "fieldsGroup": [
                {
                    "name": "detail",
                    "icon": "catalogue",
                    "position": "left",
                    "fields": [
                        {
                            "name": "name",
                            "component": "Select",
                            "componentAttributes": {
                                "translateLabels": true,
                                "options": {
                                    "scope": "local",
                                    "values": [
                                        {
                                            "label": "common.status.active",
                                            "value": 1
                                        },
                                        {
                                            "label": "common.status.inactive",
                                            "value": 0
                                        }
                                    ]
                                }
                            },
                            "validations": [
                                [
                                    {
                                        "name": "required"
                                    }
                                ],
                                [
                                    {
                                        "name": "maxLength",
                                        "options": {
                                            "length": 50
                                        }
                                    }
                                ]
                            ]
                        },
                        {
                            "name": "description",
                            "component": "Textarea",
                            "validations": [
                                [
                                    {
                                        "name": "maxLength",
                                        "options": {
                                            "length": 50
                                        }
                                    }
                                ]
                            ],
                            "componentAttributes": {}
                        },
                        {
                            "name": "appliesToLogistics",
                            "component": "Switch",
                            "validations": [
                                [
                                    {
                                        "name": "required"
                                    }
                                ]
                            ],
                            "componentAttributes": {}
                        },
                        {
                            "name": "missingStockDefault",
                            "component": "Switch",
                            "validations": [
                                [
                                    {
                                        "name": "required"
                                    }
                                ]
                            ],
                            "componentAttributes": {}
                        },
                        {
                            "name": "test",
                            "component": "Chip",
                            "componentAttributes": {
                                "borderColor": "grey"
                            }
                        },
                        {
                            "name": "user",
                            "component": "UserChip",
                            "componentAttributes": {}
                        }
                    ]
                },
                {
                    "name": "others",
                    "fields": [
                        {
                            "name": "status",
                            "component": "Select",
                            "validations": [
                                [
                                    {
                                        "name": "required"
                                    }
                                ]
                            ],
                            "componentAttributes": {
                                "translateLabels": true,
                                "options": {
                                    "scope": "local",
                                    "values": [
                                        {
                                            "label": "common.status.active",
                                            "value": 1
                                        },
                                        {
                                            "label": "common.status.inactive",
                                            "value": 0
                                        }
                                    ]
                                }
                            }
                        }
                    ]
                }
            ]
        },
        {
            "name": "semaphoreBrowse",
            "rootComponent": "BrowseSection",
            "source": {
                "service": "sac",
                "namespace": "claim-semaphore",
                "method": "browse",
                "resolve": false
            },
            "staticFilters": [
                {
                    "name": "status",
                    "target": "path",
                    "value": {
                        "dynamic": "statusId"
                    }
                },
                {
                    "name": "status",
                    "target": "query",
                    "value": {
                        "static": 1
                    }
                },
                {
                    "name": "status",
                    "target": "query",
                    "value": {
                        "static": {
                            "id": "string"
                        }
                    }
                }
            ],
            "topComponents": [
                {
                    "component": "TestComponent",
                    "attributes": {
                        "name": "test",
                        "sarasa": "test23"
                    }
                },
                {
                    "component": "TestComponent2",
                    "attributes": {
                        "name": "test",
                        "sarasa": "test23"
                    }
                }
            ],
            "fields": [
                {
                    "name": "id",
                    "component": "BoldText",
                    "attributes": {
                        "sortable": true,
                        "isDefaultSort": true,
                        "isStatus": false,
                        "initialSortDirection": "desc"
                    },
                    "mapper": "addHashtag",
                    "componentAttributes": {
                        "fontWeight": "bold"
                    }
                },
                {
                    "name": "color",
                    "component": "Text",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "fontWeight": "normal"
                    }
                },
                {
                    "name": "user",
                    "component": "UserChip",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {}
                },
                {
                    "name": "name",
                    "component": "MediumText",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "fontWeight": "medium"
                    }
                },
                {
                    "name": "rangeLow",
                    "component": "Chip",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "icon": "icon_test",
                        "iconColor": "red",
                        "borderColor": "red"
                    }
                },
                {
                    "name": "rangeHigh",
                    "component": "Chip",
                    "attributes": {
                        "isStatus": false,
                        "sortable": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "borderColor": "grey"
                    }
                },
                {
                    "name": "alerts",
                    "component": "Chip",
                    "attributes": {
                        "sortable": true,
                        "isStatus": false,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "mapper": "booleanToWord",
                    "filter": {
                        "component": "Select",
                        "componentAttributes": {
                            "translateLabels": true,
                            "options": [
                                {
                                    "label": "common.boolean.yes",
                                    "value": 1
                                },
                                {
                                    "label": "common.boolean.no",
                                    "value": 0
                                }
                            ]
                        },
                        "type": "equal",
                        "remote": false
                    },
                    "componentAttributes": {
                        "borderColor": "grey"
                    }
                },
                {
                    "name": "status",
                    "component": "StatusChip",
                    "attributes": {
                        "isStatus": true,
                        "sortable": true,
                        "isDefaultSort": false,
                        "initialSortDirection": "desc"
                    },
                    "componentAttributes": {
                        "colorSource": "statusColor"
                    },
                    "mapper": "translate",
                    "filter": {
                        "component": "Select",
                        "componentAttributes": {
                            "translateLabels": true,
                            "options": [
                                {
                                    "label": "common.status.active",
                                    "value": 1
                                },
                                {
                                    "label": "common.status.inactive",
                                    "value": 0
                                }
                            ]
                        },
                        "type": "equal",
                        "remote": false
                    }
                }
            ],
            "hasPreview": false,
            "canEdit": true,
            "canView": false,
            "pageSize": 60,
            "actions": []
        }
    ]
}